src/renderer/components/Settings/SettingsMenu.js

```diff
@@ -0,0 +1,37 @@
+// @ts-check
+import * as React from 'react'
+import Tabs from '@material-ui/core/Tabs'
+import Tab from '@material-ui/core/Tab'
+import { useIntl } from 'react-intl'
+
+/** @typedef {{tabId:import('.').SettingTabId, icon:(string | React.ReactElement<any, string | React.JSXElementConstructor<any>>), label:import('react-intl').MessageDescriptor}} SettingsTabs */
+
+/**
+ * @typedef SettingMenuProp
+ * @prop {SettingsTabs[]} tabs
+ * @prop {SettingsTabs['tabId'] | false} currentTab
+ * @prop {React.Dispatch<React.SetStateAction<false | SettingsTabs['tabId']>>} setCurrentTab
+ */
+
+/** @param {SettingMenuProp} props */
+export const SettingsMenu = ({ tabs, currentTab, setCurrentTab }) => {
+  const { formatMessage: t } = useIntl()
+
+  return (
+    <Tabs
+      orientation='vertical'
+      value={currentTab}
+      onChange={(e, newValue) => setCurrentTab(newValue)}
+    >
+      {tabs.map(tab => (
+        <Tab
+          style={{ textTransform: 'none' }}
+          label={t(tab.label)}
+          key={tab.tabId}
+          value={tab.tabId}
+          icon={tab.icon}
```

Comment:
Icon and label should be vertically aligned with each other:

![image](https://user-images.githubusercontent.com/18542095/217067330-3f7acebb-3f7d-433b-9773-1ce20a51ef6c.png)

Reply:
https://github.com/digidem/mapeo-desktop/pull/678/commits/c607d027aab4e6b44a0b17e6e74cb57fba736a3e

Reply:
for what its worth, this was annoyingly hard to do with MUI ..... :joy: